subgraphs/uniswap-v2/src/core.ts

```diff
@@ -0,0 +1,527 @@
+  let token0Amount = convertTokenToDecimal(event.params.amount0, token0.decimals)
+  let token1Amount = convertTokenToDecimal(event.params.amount1, token1.decimals)
+
+  let ether = _HelperStore.load('ETH')
+  if (ether == null) return
+
+  let token0USD = tokenTracker0.derivedETH.times(ether.valueDecimal!)
+  let token1USD = tokenTracker1.derivedETH.times(ether.valueDecimal!)
+
+  let withdrawal = new Withdraw(
+    event.transaction.hash
+      .toHexString()
+      .concat('-')
+      .concat(event.logIndex.toString())
+  )
+
+  withdrawal.hash = event.transaction.hash.toHexString()
+  withdrawal.logIndex = event.logIndex.toI32()
+  withdrawal.protocol = protocol.id
+  withdrawal.to = event.params.to.toHexString()
+  withdrawal.from = event.params.sender.toHexString()
+  withdrawal.blockNumber = event.block.number
+  withdrawal.timestamp = event.block.timestamp
+  withdrawal.inputTokens = [pool.inputTokens[0], pool.inputTokens[1]]
+  withdrawal.outputTokens = pool.outputToken
+  withdrawal.inputTokenAmounts = [token0Amount, token1Amount]
+  withdrawal.outputTokenAmount = burn.liquidity
+  withdrawal.amountUSD = token0USD.times(token0Amount).plus(token1USD.times(token1Amount))
+
+
+  updateUsageMetrics(event.block.number, event.block.timestamp, event.transaction.from)
+  updateFinancials(event.block.number, event.block.timestamp)
+  updatePoolMetrics(event.block.number, event.block.timestamp, pool)
+  
+  burn.save()
+  withdrawal.save()
+  pool.save()
+}
+
+
+
+export function handleSwap(event: Swap): void {
+  let pool = LiquidityPool.load(event.address.toHexString())
+  if (pool == null) return
+
+  let protocol = DexAmmProtocol.load(FACTORY_ADDRESS)
+  if(protocol == null) return
+
+  let token0 = Token.load(pool.inputTokens[0])
+  let token1 = Token.load(pool.inputTokens[1])
+  let tokenTracker0 = _TokenTracker.load(pool.inputTokens[0])
+  let tokenTracker1 = _TokenTracker.load(pool.inputTokens[1])
+  
+  if (token0 == null) return
+  if (token1 == null) return
+  if (tokenTracker0 == null) return
+  if (tokenTracker1 == null) return
+
+
+  let amount0In = convertTokenToDecimal(event.params.amount0In, token0.decimals)
+  let amount1In = convertTokenToDecimal(event.params.amount1In, token1.decimals)
+  let amount0Out = convertTokenToDecimal(event.params.amount0Out, token0.decimals)
+  let amount1Out = convertTokenToDecimal(event.params.amount1Out, token1.decimals)
+
+  // totals for volume updates
+  let amount0Total = amount0Out.plus(amount0In)
+  let amount1Total = amount1Out.plus(amount1In)
+
+  // ETH/USD prices
+  let ether = _HelperStore.load('ETH')
+  if (ether == null) return
+
+  let token0USD = tokenTracker0.derivedETH.times(amount0Total).times(ether.valueDecimal!)
+  let token1USD = tokenTracker1.derivedETH.times(amount1Total).times(ether.valueDecimal!)
+
+  /// get total amounts of derived USD for tracking
+  let derivedAmountUSD = token1USD.plus(token0USD).div(BigDecimal.fromString('2'))
```

Comment:
Save `BigDecimal.fromString('2')` as a constant so we can reuse

Reply:
Fixed. But I also commented out this line of code since it is not used.